Brokerages/Backtesting/BacktestingBrokerage.cs

```diff
@@ -318,34 +319,22 @@ public virtual void Scan()
                     {
                         //Model:
                         var model = security.FillModel;
-
                         //Based on the order type: refresh its model to get fill price and quantity
                         try
                         {
                             switch (order.Type)
                             {
                                 case OrderType.Limit:
-                                    fills = new[] { model.LimitFill(security, order as LimitOrder) };
-                                    break;
-
                                 case OrderType.StopMarket:
-                                    fills = new[] { model.StopMarketFill(security, order as StopMarketOrder) };
-                                    break;
-
                                 case OrderType.Market:
-                                    fills = new[] { model.MarketFill(security, order as MarketOrder) };
-                                    break;
-
                                 case OrderType.StopLimit:
-                                    fills = new[] { model.StopLimitFill(security, order as StopLimitOrder) };
-                                    break;
-
                                 case OrderType.MarketOnOpen:
-                                    fills = new[] { model.MarketOnOpenFill(security, order as MarketOnOpenOrder) };
-                                    break;
-
                                 case OrderType.MarketOnClose:
-                                    fills = new[] { model.MarketOnCloseFill(security, order as MarketOnCloseOrder) };
+                                    var context = new FillModelContext(
```

Comment:
in order to better support polymorphism, we should make the fill model the default and do:
``` csharp
if (order.Type == OrderType.OptionExercise) // do opt exercise
else { // invoke fill model for everything else }
```
As it is now, if we (or a user) added a new order type it, changes would be needed here in order to support it, just so it goes to the fill model. This should happen automatically.